kahuna/public/js/search/query.js

```diff
@@ -62,11 +56,13 @@ query.controller('SearchQueryCtrl',
     Object.keys($stateParams)
           .forEach(setAndWatchParam);
 
-    // pass undefined to the state on empty to remove the QueryString
-    function valOrUndefined(str) { return str || undefined; }
+    // URL parameters are not decoded when taken out of the params.
+    // Might be fixed with: https://github.com/angular-ui/ui-router/issues/1759
+    // Pass undefined to the state on empty to remove the QueryString
+    function valOrUndefined(str) { return str ? decodeURIComponent(str) : undefined; }
```

Comment:
`setAndWatchParam` implies that - we need to do this on load, then when it changes. Basically whenever we set it.
I would argue the `setAndWatchParam` paradigm could use a little work, but we do want it in both places here.